os/src/FileOps.scala

```diff
@@ -38,6 +42,7 @@ object makeDir extends Function1[Path, Unit] {
   object all extends Function1[Path, Unit] {
     def apply(path: Path): Unit = apply(path, null, true)
     def apply(path: Path, perms: PermSet = null, acceptLinkedDirectory: Boolean = true): Unit = {
+      checker.value.onWrite(path)
```

Comment:
The original implementation creates any necessary enclosing directories via `java.nio.file.Files#createDirectories`.  

Here the checker is only checking for write access in the full path.

Don't we need to apply it to the enclosing directories as well?

I see this would entail providing a shim for `java.nio.file.Files#createDirectories`.